packages/yarnpkg-core/sources/miscUtils.ts

```diff
@@ -240,3 +240,7 @@ export function sortMap<T>(values: Iterable<T>, mappers: ((value: T) => string)
     return asArray[index];
   });
 }
+
+export function getCompressionLevel(linker: string | null, compressionLevel: number): number {
+  return compressionLevel >= 0 ? compressionLevel : linker === 'node-modules' ? 0 : -1;
```

Comment:
I'm not sure we want to do that by default, even on the n_m linker.

- It trades size for speed, and it's not clear whether one is better than the other in a general case (size can be seen as a recurring cost, whereas fetch time could be a one time cost).
- It requires refetching the entire cache each time you switch between the nm and the PnP linker, which might hinder the adoption of one or the other.

Imo we should make the default cache the same for both linkers.

Reply:
This function is unused, I forgot to remove it. Done. The defaults are the same for both linkers